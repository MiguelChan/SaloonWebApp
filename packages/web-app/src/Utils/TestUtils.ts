import { ProductStep, ProductValue } from "@mchanc/shared-library";

export const renderProductValues = (numberOfElements: number): ProductValue[] => {
  const productValues: ProductValue[] = [];
  for (let i = 0; i < numberOfElements; i++) {
    const productValue = {
      imageUrl: 'https://source.unsplash.com/random',
      altTitle: `Test + ${i}`,
      title: `Another Title: ${i}`,
      description: `This is some description for ${i}`,
    };
    productValues.push(productValue);
  }
  return productValues;
}

export const renderProductSteps = (numberOfElements: number): ProductStep[] => {
  const productSteps: ProductStep[] = [];
  for (let i = 0; i < numberOfElements; i++) {
    const productStep: ProductStep = {
      stepImage: 'https://source.unsplash.com/random',
      stepNumber: i,
      stepDescription: `This is some description for ${i}`,
    };
    productSteps.push(productStep);
  }
  return productSteps;
}