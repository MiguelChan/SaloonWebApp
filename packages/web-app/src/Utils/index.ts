export * from './Validation';
export { default as theme } from './Theme';