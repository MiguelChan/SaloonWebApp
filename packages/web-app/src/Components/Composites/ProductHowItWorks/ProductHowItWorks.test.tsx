import React from 'react';
import {
  render,
} from '@testing-library/react';
import { ProductHowItWorks, ProductHowItWorksProps } from '.';
import { ThemeProvider } from '@mui/styles';
import { theme } from 'Utils';
import { renderProductSteps } from 'Utils/TestUtils';

describe('ProductHowItWorks', () => {

  it('Should match the Snapshot', () => {
    const props: ProductHowItWorksProps = {
      productSteps: renderProductSteps(3),
    };

    const {
      container,
    } = renderComponent(props);

    expect(container).toMatchSnapshot();
  });

  const renderComponent = (props: ProductHowItWorksProps) => {
    const Component = (
      <ThemeProvider theme={theme}>
        <ProductHowItWorks {...props} />
      </ThemeProvider>
    );

    return render(Component);
  };
});