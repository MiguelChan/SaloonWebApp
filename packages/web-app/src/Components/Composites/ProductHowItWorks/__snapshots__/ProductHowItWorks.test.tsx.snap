// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductHowItWorks Should match the Snapshot 1`] = `
<div>
  <section
    class="MuiBox-root css-1lv7ehy"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1llydmb-MuiContainer-root"
    >
      <img
        alt="curvy lines"
        class="MuiBox-root css-17w9039"
        src="productCurvyLines.png"
      />
      <h2
        class="MuiTypography-root MuiTypography-h4 css-1amj3j2-MuiTypography-root"
      >
        How it works
        <span
          class="Typography-markedH4Center-3"
        />
      </h2>
      <div>
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 css-v9705b-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
            >
              <div
                class="MuiBox-root css-19zp38f"
              >
                <div
                  class="MuiBox-root css-sjgcuh"
                >
                  0
                  .
                </div>
                <img
                  alt="This is some description for 0"
                  class="MuiBox-root css-3ozjgl"
                  src="https://source.unsplash.com/random"
                />
                <h3
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-ph2xsj-MuiTypography-root"
                >
                  This is some description for 0
                </h3>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
            >
              <div
                class="MuiBox-root css-19zp38f"
              >
                <div
                  class="MuiBox-root css-sjgcuh"
                >
                  1
                  .
                </div>
                <img
                  alt="This is some description for 1"
                  class="MuiBox-root css-3ozjgl"
                  src="https://source.unsplash.com/random"
                />
                <h3
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-ph2xsj-MuiTypography-root"
                >
                  This is some description for 1
                </h3>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
            >
              <div
                class="MuiBox-root css-19zp38f"
              >
                <div
                  class="MuiBox-root css-sjgcuh"
                >
                  2
                  .
                </div>
                <img
                  alt="This is some description for 2"
                  class="MuiBox-root css-3ozjgl"
                  src="https://source.unsplash.com/random"
                />
                <h3
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-ph2xsj-MuiTypography-root"
                >
                  This is some description for 2
                </h3>
              </div>
            </div>
          </div>
        </div>
      </div>
      <a
        class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root css-q34hpo-MuiButtonBase-root-MuiButton-root"
        href="/premium-themes/onepirate/sign-up/"
        tabindex="0"
      >
        Get started
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </section>
</div>
`;
